packages/core/src/utils.ts

```diff
@@ -168,20 +168,21 @@ export const getGasPriceOverrides = async (
       executionMode === ExecutionMode.Platform) &&
     process.env.SPHINX_INTERNAL__DISABLE_HARDCODED_GAS_LIMIT !== 'true'
   ) {
-    // Hard-code the gas limit to be the 3/4 of the block gas limit. This is an optimization that
-    // significantly speeds up deployments because it removes the need for EthersJS to call
-    // `eth_estimateGas`, which is a very slow operation for large transactions. We only make this
-    // optimization in situations where we can safely assume that the caller has an unlimited amount
-    // of ETH. We don't override this on live networks because the signer is the user's wallet,
-    // which may have a limited amount of ETH. We set it to 3/4 of the block gas limit for two
-    // reasons:
-    // 1. This value is considerably higher than the max batch size for `EXECUTE` Merkle leaves,
-    //    which ensures that we don't accidentally underfund the transaction.
-    // 2. This value is considerably lower than the block gas limit. If, instead, we set this value
-    //    equal to the current block gas limit, a situation could occur where the block gas limit
-    //    decreases slightly after we set this value, which would cause an error due to the fact
-    //    that the transaction's gas limit exceeds the block gas limit.
-    overridden.gasLimit = (BigInt(3) * block.gasLimit) / BigInt(4)
+    // Get the max batch gas limit on the current network.
+    const maxGasLimit = getMaxGasLimit(block.gasLimit, network.chainId)
+    // Hard-code the gas limit to be midway between the max batch gas limit and the block gas limit.
+    // This is an optimization that significantly speeds up deployments because it removes the need
+    // for EthersJS to call `eth_estimateGas`, which is a very slow operation for large
+    // transactions. We only make this optimization in situations where we can safely assume that
+    // the caller has an unlimited amount of ETH. We don't override this on live networks because
+    // the signer is the user's wallet, which may have a limited amount of ETH.
+    //
+    // We set this value higher than the max batch gas limit to ensure that we don't accidentally
+    // underfund the transaction. If we set this value equal to the block gas limit, a situation
+    // could occur where the block gas limit decreases slightly after we set this value, which would
+    // cause an error due to the fact that the transaction's gas limit exceeds the block gas limit.
+    // This occurred when simulating a deployment on Polygon.
+    overridden.gasLimit = (maxGasLimit + block.gasLimit) / BigInt(2)
```

Comment:
I think this is preferable to hardcoding `overridden.gasLimit`to something like 95% because this strategy ensures that we won't run into this bug again, even if we set a max batch size that's _really_ close to the block gas limit.